main.py

```diff
@@ -115,26 +115,30 @@ def process(data, factors):
             average[factor_set][surface] = np.average(np.array(average[factor_set][surface]), axis=0)
     return average
 
-def sort_directions(data):
+def choose_direction(data, average):
     user_surfaces = []
     for direction_name, surface_arrays in data.items():
-        user_surfaces.append((direction_name, len(surface_arrays.keys()),surface_arrays.keys()))
-    return sorted(user_surfaces, key=lambda x: x[1], reverse=True)
-
+        user_surfaces.append((direction_name, len(surface_arrays.keys())))
+    #get the number of surfaces stored in average as keys
+    user_surfaces = sorted(user_surfaces, key=lambda x: x[1], reverse=True)
```

Comment:
Line 119-123 can be compressed into a single line:
```user_surfaces = sorted([(direction, len(surfaces)) for direction, surfaces in data.items()], key=lambda x: x[1],reverse=True)```

Reply:
Hoooo....one line ko three lines

Reply:
@harrywhoforks Please change it in the next commit~

Reply:
@harrywhoforks I can see the new version doesn't sort the directions by their user surfaces any more, I will mark this as resolved.